scripts/heat/instance_pairs.yaml

```diff
@@ -0,0 +1,186 @@
+    default: 172.20.1.99
+
+resources:
+
+  private_net_east:
+    type: OS::Neutron::Net
+    properties:
+        name: { get_param: east_net_name }
+
+  private_subnet_east:
+    type: OS::Neutron::Subnet
+    properties:
+      network_id: { get_resource: private_net_east }
+      cidr: { get_param: east_net_cidr }
+      allocation_pools:
+        - start: { get_param: east_net_pool_start }
+          end: { get_param: east_net_pool_end }
+
+  private_net_west:
+    type: OS::Neutron::Net
+    properties:
+        name: { get_param: west_net_name }
+
+  private_subnet_west:
+    type: OS::Neutron::Subnet
+    properties:
+      network_id: { get_resource: private_net_west }
+      cidr: { get_param: west_net_cidr }
+      allocation_pools:
+        - start: { get_param: west_net_pool_start }
+          end: { get_param: west_net_pool_end }
+
+  router:
+    type: OS::Neutron::Router
+    properties:
+      external_gateway_info:
+        network: { get_param: public_net }
+
+  router_interface_east:
+    type: OS::Neutron::RouterInterface
+    properties:
+      router_id: { get_resource: router }
+      subnet_id: { get_resource: private_subnet_east }
+
+  router_interface_west:
+    type: OS::Neutron::RouterInterface
+    properties:
+      router_id: { get_resource: router }
+      subnet_id: { get_resource: private_subnet_west }
+
+  security_group:
+    type: OS::Neutron::SecurityGroup
+    properties:
+      rules:
+        - direction: ingress
+          ethertype: IPv4
+
+  east_port:
+    type: OS::Neutron::Port
+    properties:
+      security_groups:
+        - { get_resource: security_group }
+      network_id: { get_resource: private_net_east }
+      fixed_ips:
+        - subnet_id: { get_resource: private_subnet_east }
+
+  east_floating_ip:
+    type: OS::Neutron::FloatingIP
+    properties:
+      floating_network: { get_param: public_net }
+      port_id: { get_resource: east_port }
+
+  west_port:
+    type: OS::Neutron::Port
+    properties:
+      security_groups:
+        - { get_resource: security_group }
+      network_id: { get_resource: private_net_west }
+      fixed_ips:
+        - subnet_id: { get_resource: private_subnet_west }
+
+  west_floating_ip:
+    type: OS::Neutron::FloatingIP
+    properties:
+      floating_network: { get_param: public_net }
+      port_id: { get_resource: west_port }
+
+  east:
+    type: OS::Nova::Server
+    properties:
+      name: { get_param: east_name }
+      image: { get_param: image }
+      flavor: { get_param: flavor }
+      key_name: testkey
+      networks:
+        - port: { get_resource: east_port }
+      user_data_format: RAW
+      user_data:
+        str_replace:
+          template: { get_file: ping_template.sh }
+          params:
+            "@outside_ip@": { get_param: outside_ip }
+            "@neighbour_ip@": { get_attr: [ west_floating_ip, floating_ip_address ] }
+
+  west:
+    type: OS::Nova::Server
+    properties:
```

Comment:
I'm not sure I'm happy with this as it is. Depending on the cloud's load level the instances might all end up on the same host. I'd rather create a [ServerGroup](https://docs.openstack.org/developer/heat/template_guide/openstack.html#OS::Nova::ServerGroup) in the parent template (with the default anti-affinity constraint that comes with) and pass a matching `scheduler_hint` parameter here (see the [Nova documentation](https://docs.openstack.org/developer/heat/template_guide/openstack.html#OS::Nova::ServerGroup) for what the payload for that looks like - unfortunately Heat only has bare bones support for this parameter so far).

Can you try if this works please (it's a little used feature and as such it may well be broken in Heat, but let's at least try it)? Thanks!